### PR TITLE
Accepting answers for questions

### PR DESCRIPTION
**Pull request by ananyo2012**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] tests pass -- `rake test`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request are descriptively named
* [x] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [ ] `development.sqlite.example` has been updated if any database migrations were added

This implements the feature for accepting answer for questions. The user can accept answers for questions of which he is an author. Screenshots given below.

Thanks!

### CODE REVIEW

**Comment by ananyo2012** [comment]
Here is how a accepted answer and a normal answer looks when user is not logged in
accept_answer_v1a
Here is how it looks to  user who is logged in and is the author of the question. On clicking the **Accepted** button a green colored label appears and on clicking over the button again the answer is unaccepted again an the colour becomes grey
accept_answer_v1b

**Comment by ananyo2012** [comment]
Currently the author of the question can accept multiple answers at a time. I think it is more appropriate. But would it be more correct to allow only one answer to be accepted? If so I can make the necessary changes.
@jywarren Please check when you will be back.

**Comment by dhenrygithub** [comment]
Ananyo,
That looks good to me - just one suggestion:
Change the text of the unchecked "accepted" button in grey to "accept" to more exactly match the verb it implements.  This also makes it more clear that only the accepted answer(s) actually say "accepted".

**Comment by ananyo2012** [comment]
Updated the text as per @dhenrygithub. The 'accept' button will change to 'accepted' button on clicking and vice-versa. Here is the screenshot
answer_accept_v2

**Comment by jywarren** [comment]
This looks great -- a few small stylistic followup requests:

* could we make the share link just a share icon in a button (like the Likes button)? http://fontawesome.io/icon/share-alt/ -- this will also make the pipe `|` character unnecessary
* the link for Comments should also include the # of comments, like [4 comments](). 
* I think the Accept button should go at the start of the line, and be white like the Like button. And say `Accept this answer` just to be extra clear. 

I know these are very small changes, but i think if we get them right, it'll look fantastic! Great work and thanks for continuing while I was gone!

**Comment by ananyo2012** [comment]
@jywarren Made the style tweaks. Though I didn't complete get your second point. The no of comments was already present. But I lso included the number in the link. Here is the screenshot
answer_accept_v3

**Comment by jywarren** [comment]
Sorry yes that's perfect -- i just meant the number should be in the link, so thanks! Trying a merge now.

File: db/migrate/20160629044639_add_accepted_field_to_answers.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class AddAcceptedFieldToAnswers < ActiveRecord::Migration
+  def up
+  	add_column :answers, :accepted, :boolean, default: false
```
Comment (jywarren) [inline db/migrate/20160629044639_add_accepted_field_to_answers.rb]: Ah - two space tabs here please. I'll fix in an extra commit this time. Thanks!

File: app/assets/stylesheets/style.css [hunk 2]
```diff
@@ -450,3 +450,11 @@ pre {
   /* Hide the text; just using it for sizing */
   visibility: hidden;
 }
+
+.answer-accept{
+  font-weight: bold;
+}
+
+.answer-accept.btn-success{
```
Comment (jywarren) [inline app/assets/stylesheets/style.css]: Sorry to nitpick; a space before the `{` here would be great; i'll fix in an extra commit this time. Thanks!

**Comment by jywarren** [comment]
Excellent; passes tests and looks good. Just merged manually with a couple tweaks as mentioned in comments above: 3d6b20f38f44c958422beb0ec20e9e6cba992dd3

**Comment by ananyo2012** [comment]
@jywarren Thanks Jeff. Sorry for the coding styles. Missed it again

**Pull request merged**